platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -1266,30 +1276,24 @@ private <R, S> void setMapSnapshotAndArray(
      * @throws IOException
      * 		If something fails.
      */
-    private void serializePathToDirtyInternalIndex(
-            final Map<Long, Mutation<VirtualInternalRecord>> map, final SerializableDataOutputStream out)
-            throws IOException {
+    private void serializePathToDirtyHashIndex(
+            final Map<Long, Mutation<Long, Hash>> map, final SerializableDataOutputStream out) throws IOException {
         assert snapshot.get() : "Only snapshots can be serialized";
         out.writeInt(map.size());
-        for (final Map.Entry<Long, Mutation<VirtualInternalRecord>> entry : map.entrySet()) {
+        for (final Map.Entry<Long, Mutation<Long, Hash>> entry : map.entrySet()) {
             out.writeLong(entry.getKey());
-            final Mutation<VirtualInternalRecord> mutation = entry.getValue();
+            final Mutation<Long, Hash> mutation = entry.getValue();
             assert mutation != null : "Mutations cannot be null in a snapshot";
             assert mutation.version <= this.fastCopyVersion.get()
                     : "Trying to serialize pathToDirtyInternalIndex with a version ahead";
-
-            final VirtualInternalRecord internalRecord = mutation.value;
             out.writeLong(mutation.version);
             out.writeBoolean(mutation.deleted);
-            if (internalRecord != null) {
-                out.writeLong(internalRecord.getPath());
-                out.writeSerializable(internalRecord.getHash(), true);
-            }
+            out.writeSerializable(mutation.value, true);
```

Comment:
This is incompatible. Even if mutation.value is guaranteed never to be null, the path is still not written as before (and not read on deserialize). How will existing snapshots be loaded from previous versions, then? Note that it's different from data source stores, virtual node cache records are serialized directly to the state file.

Reply:
Yes, thank you for catching that. I'll introduce another `ClassVersion` for `VirtualNodeCache`.